# The complete makefile will run to the compile step
all : build

# Remove previous processing output
clean :
	rm -r output

# Identify script directory
source = code

# Build conda environments
renv = renv/library
build : $(renv)
$(renv) : $(source)/build.sh
	bash -i $<

# Download reads
# This will pull reads off of the CQLS HPCC

# Demultiplex reads
# These targets will ultimately need to call "conda run" to be produced
demux_out = output/demux
demux : build $(demux.out)
$(demux_out) : $(source)/pheniqs.config.json
	mkdir -p $(demux.out)
	pheniqs mux -R $(demux.out)/demux.report.txt -c $< --base-input data/raw --base-output $(demux.out)

# Trim primers + adapters with cutadapt and SeqPurge
trim_out = output/trim
trim : demux $(trim_out)
$(trim_out) : $(source)/trim.sh
	$<

# Denoise with dada2
denoise_out = output/denoise/seq.tab.rds
denoise : trim $(denoise_out)
$(denoise_out) : $(source)/denoise.R
	Rscript $<

# Trim with ITSx, remove chimeras, identify fungal/non-fungal sequences,
# post-cluster with LULU, assign taxonomy, filter with PERFect,
# and save the output as an rds file

# We will be investigating DIAMOND + MEGAN for more robust taxonomy assignment.

compile.out = output/compile/phy.list.rds
compile : denoise $(compile_out)
$(compile_out) : $(source)/compile.R
	Rscript $<

# We also need to generate an rmd report to summarize the sequencing run.
# This should include:
# - Read counts after each primer + adapter trimming step
# - Pre- and post-trimmed/filtered quality profile graphs produced with dada2
# - Error plots generated by dada2
# - Sequence counts after each step in denoise.R
